pkg/container/network.go

```diff
@@ -237,6 +233,44 @@ func addLink(link netlink.Link) (err error) {
 	return
 }
 
+// This is a MAC address persistence workaround, netlink.LinkSetMaster{,ByIndex}()
+// has a bug that arbitrarily changes the MAC addresses of the bridge and virtual
+// device to be bound to it. TODO: Remove when fixed upstream
+func setMaster(master netlink.Link, links ...netlink.Link) error {
+	masterIndex := master.Attrs().Index
+	masterMAC, err := getMAC(master)
+	if err != nil {
+		return err
+	}
+
+	for _, link := range links {
+		mac, err := getMAC(link)
+		if err != nil {
+			return err
+		}
+
+		if err = netlink.LinkSetMasterByIndex(link, masterIndex); err != nil {
+			return err
+		}
+
+		if err = netlink.LinkSetHardwareAddr(link, mac); err != nil {
+			return err
+		}
+	}
+
+	return netlink.LinkSetHardwareAddr(master, masterMAC)
+}
+
+// getMAC fetches the generated MAC address for the given link
+func getMAC(link netlink.Link) (addr net.HardwareAddr, err error) {
+	if link, err = netlink.LinkByIndex(link.Attrs().Index); err != nil {
```

Comment:
Why are you "overwriting" this variable?

Reply:
The passed link is the "constructor" and has no hardware address since we want it to be generated by the network subsystem, so I need to "reload" the link by requesting the properties from the system after it has been created, which fills all the fields. I'll add a comment to avoid confusion.